src/main/java/mate/academy/spring/validation/EqualFieldsValidator.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.spring.validation;
+
+import java.lang.reflect.Field;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EqualFieldsValidator implements ConstraintValidator<EqualPasswords, Object> {
+    private static final String VALUE_IS_NULL = "Passwords must not be null";
+    private static final String VALUES_NOT_EQUAL = "Passwords are not equal";
+    private String password;
+    private String repeatPassword;
+
+    @Override
+    public void initialize(EqualPasswords constraintAnnotation) {
+        password = constraintAnnotation.firstValue();
+        repeatPassword = constraintAnnotation.secondValue();
+    }
+
+    @Override
+    public boolean isValid(Object clazz, ConstraintValidatorContext constraintValidatorContext) {
+        constraintValidatorContext.disableDefaultConstraintViolation();
+        Object valueOne = getValue(clazz, password);
+        Object valueTwo = getValue(clazz, repeatPassword);
+        if (valueOne == null || valueTwo == null) {
+            customMessage(constraintValidatorContext, VALUE_IS_NULL);
+            return false;
+        }
+        if (!valueOne.equals(valueTwo)) {
+            customMessage(constraintValidatorContext, VALUES_NOT_EQUAL);
+            return false;
+        }
```

Comment:
Just a hint) 
You can compare your passwords with nullsafe Objects.equals(firstParam, secondParam)

Reply:
the idea was if one of obkects is null so it will be send VALUE_IS_NULL custom message, if objects not equal so it will be send another custom message VALUE_NOT_EQUAL

Reply:
Got it, good idea